webapp/src/main/java/org/apache/atlas/web/rest/TypesREST.java

```diff
@@ -455,7 +455,7 @@ public AtlasTypesDef createAtlasTypeDefs(@Context HttpServletRequest servletRequ
                                            @QueryParam("allowCustomName") @DefaultValue("false") boolean allowCustomName) throws AtlasBaseException {
         Lock lock = null;
         AtlasPerfTracer perf = null;
-        validateTypeCreateOrUpdate(typesDef);
+        validateTypeBeforeAction(typesDef);
```

Comment:
This will only handle subsequent type creation requests which are dependent on the type created earlier

What about scenarios like
1. Create a CM attributeDef
2. Immediately update an asset to link new CM which goes to another pod where typeDefs cache is not refreshed